pkg/reconciler/knativeeventing/common/dispatcheradaptertransform.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+type unstructuredGetter interface {
+	Get(obj *unstructured.Unstructured) (*unstructured.Unstructured, error)
+}
+
+// DispatcherAdapterTransform keeps the number of replicas and the env vars, if the deployment
+// pingsource-mt-adapter, kafka-ch-dispatcher or imc-dispatcher exists in the cluster.
+func DispatcherAdapterTransform(client unstructuredGetter) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() == "Deployment" && (u.GetName() == "pingsource-mt-adapter" ||
+			u.GetName() == "imc-dispatcher") {
```

Comment:
do we really want the `imc-dispatcher` here ?  (title of this is: Add the transformer for pingsource). and perhaps we do more focused/smaller PRs? /cc @antoineco

Reply:
we also seem to lack code for `imc-dispatcher` test?  IMO let's do it on a separate PR ?

Reply:
Like I said, I'm totally fine with that 👍

Reply:
yueah, let's rmeove the imc-dispatcher

Reply:
ok